### PR TITLE
use defaults from futureDefaults

### PR DESCRIPTION
**Pull request by vankop**
**What kind of change does this PR introduce?**

feature

**Did you add tests for your changes?**

no

**Does this PR introduce a breaking change?**

yes

**What needs to be documented once your changes are merged?**

new defaults

### CODE REVIEW

File: test/__snapshots__/StatsTestCases.basictest.js.snap [hunk 1]
```diff
@@ -193,6 +193,13 @@ exports[`StatsTestCases should print correct stats for async-commons-chunk-auto
     > ./a ./index.js 1:0-47
     dependent modules 60 bytes [dependent] 3 modules
     ./a.js + 1 modules 185 bytes [built] [code generated]
+
+  LOG from webpack.FileSystemInfo
+  <w> Managed item Xdir/async-commons-chunk-auto/node_modules/z.js isn't a directory or doesn't contain a package.json
```
Comment (vankop) [inline test/__snapshots__/StatsTestCases.basictest.js.snap:198]: I'll change to correct one

Comment (sokra) [reply test/__snapshots__/StatsTestCases.basictest.js.snap:198]: Maybe move the file to `node_modules/z/index.js` and add a single `package.json` with `{ "name": "z", version: "1.0.0" }`.

File: lib/config/defaults.js [hunk 2]
```diff
@@ -988,26 +928,19 @@ const applyLoaderDefaults = (loader, { targetProperties }) => {
...
 		if (targetProperties && targetProperties.node) return "eval-only";
-		// TODO webpack 6 should always default to false
-		return futureDefaults ? "warn-mock" : "mock";
+		return false;
 	});
 	F(node, "__dirname", () => {
 		if (targetProperties && targetProperties.node) return "eval-only";
-		// TODO webpack 6 should always default to false
-		return futureDefaults ? "warn-mock" : "mock";
+		return false;
```
Comment (alexander-akait) [inline lib/config/defaults.js:943]: I think will be good to add these changes in readme and how to return previously behavior

File: test/cases/scope-hoisting/json-reexport-6700/json.js [hunk 3]
```diff
@@ -4,8 +4,10 @@ export { aa };
...
 export { default as bb } from './b.json';
 
-import { a } from './a.json';
+import bbb from './b.json';
 
-export { a };
+const a = aa.a;
+const b = bbb.b;
+
+export { a, b };
```
Comment (sokra) [inline test/cases/scope-hoisting/json-reexport-6700/json.js:12]: This is no longer a reexport. Instead you can reexport the default value.

File: test/WatchTestCases.template.js [hunk 4]
```diff
@@ -141,6 +141,9 @@ const describeCases = config => {
 									if (!options.context) options.context = tempDirectory;
 									if (!options.entry) options.entry = "./index.js";
 									if (!options.target) options.target = "async-node";
+									if (!options.module) options.module = {};
+									if (!("unsafeCache" in options.module))
+										options.module.unsafeCache = false;
```
Comment (sokra) [inline test/WatchTestCases.template.js:146]: Add a TODO here to remove that again `// TODO webpack 6 remove this after fixing unsafeCache`

File: test/cases/side-effects/dynamic-reexports/warnings.js [hunk 5]
```diff
@@ -1,5 +0,0 @@
-module.exports = [
-	[
-		/export 'default' \(imported as 'Default1'\) was not found in '\.\/direct-export'/
```
Comment (sokra) [inline test/cases/side-effects/dynamic-reexports/warnings.js:3]: The error here is important for the test. It need to check that the `default` export is not reexported from a `export *`. Could you verify with `__webpack_exports_info__.default.provided` inside of `direct-export` that it still works as expected.

File: test/fixtures/buildDependencies/node_modules/dependency-with-exports/main.js [hunk 6]
```diff
@@ -1,4 +1,2 @@
 require("dep-with-exports");
 require("dep-with-exports/sub");
-require("dep-without-package.json/main-entry");
-require("dep-without-package.json/sub-entry");
```
Comment (sokra) [inline test/fixtures/buildDependencies/node_modules/dependency-with-exports/main.js:4]: hmm.. this doesn't work anymore... I guess we should add some kind of fallback for that to no regress on that. (should be added to `main`)

Comment (sokra) [reply test/fixtures/buildDependencies/node_modules/dependency-with-exports/main.js:4]: Seems like the test wasn't working as intended, as it didn't use managedPaths for the node_modules folder...

File: test/WatchTestCases.template.js (hunk 4 above)
Comment (sokra) [reply test/WatchTestCases.template.js:146]: Or we wait for this to be fixed and remove it from the PR

**Review by sokra — CHANGES_REQUESTED** [review]

File: test/fixtures/buildDependencies/node_modules/dependency-with-exports/main.js (hunk 6 above)
Comment (vankop) [reply test/fixtures/buildDependencies/node_modules/dependency-with-exports/main.js:4]: > Seems like the test wasn't working as intended, as it didn't use managedPaths for the node_modules folder...

we can fix test case in `main` (provide `managedPaths` in `webpack.config`)

Comment (sokra) [reply test/fixtures/buildDependencies/node_modules/dependency-with-exports/main.js:4]: yep

Comment (sokra) [reply test/fixtures/buildDependencies/node_modules/dependency-with-exports/main.js:4]: fixed by #14735

File: test/WatchTestCases.template.js (hunk 4 above)
Comment (sokra) [reply test/WatchTestCases.template.js:146]: fixed by #14735

**Review by sokra — APPROVED** [review]

**Pull request merged**